contracts/0.4.24/lib/SigningKeys.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+// See contracts/COMPILERS.md
+pragma solidity 0.4.24;
+
+import {SafeMath} from "@aragon/os/contracts/lib/math/SafeMath.sol";
+import {SafeMath64} from "@aragon/os/contracts/lib/math/SafeMath64.sol";
+import {MemUtils} from "../../common/lib/MemUtils.sol";
+
+import "hardhat/console.sol";
+
+library SigningKeys {
+    using SafeMath for uint256;
+    using SafeMath64 for uint64;
+
+    using SigningKeys for bytes32;
+
+    uint64 internal constant PUBKEY_LENGTH = 48;
+    uint64 internal constant SIGNATURE_LENGTH = 96;
+    uint256 internal constant UINT64_MAX = uint256(~uint64(0));
+
+    event SigningKeyAdded(uint256 indexed nodeOperatorId, bytes pubkey);
+    event SigningKeyRemoved(uint256 indexed nodeOperatorId, bytes pubkey);
+
+    function isKeyEmpty(bytes memory _key) internal pure returns (bool) {
+        assert(_key.length == PUBKEY_LENGTH);
+
+        uint256 k1;
+        uint256 k2;
+        assembly {
+            k1 := mload(add(_key, 0x20))
+            k2 := mload(add(_key, 0x40))
+        }
+
+        return 0 == k1 && 0 == (k2 >> ((2 * 32 - PUBKEY_LENGTH) * 8));
```

Comment:
DISCLAIMER :skull: 
totally not sure that this is a good idea, but decided to try my best :crab: 
```
        bool ret;
        assembly {
            ret  := iszero(and(mload(add(_key, 0x20)), shr(mload(add(_key, 0x40)), 0x80)))
        }
 ```